### PR TITLE
Paintroid 451 : Flutter: advanced Options

### PR DESCRIPTION
**Pull request by vishad-tyagi**
Ticket
https://catrobat.atlassian.net/browse/PAINTROID-451
Paintroid-451 : Flutter: advanced Options

Description
Add the functionality for the advanced options in the overflow menu.

## New Features and Enhancements
- New Features and Enhancements
## Refactorings and Bug Fixes
- Refactorings and Bug Fixes

#### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Paintroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#paintroid* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: packages/io_library/lib/src/ui/advanced_options_dialog.dart [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+class AdvancedOptionsDialog extends StatefulWidget {
+  const AdvancedOptionsDialog({Key? key}) : super(key: key);
+
+  @override
+  AdvancedOptionsDialogState createState() => AdvancedOptionsDialogState();
+}
+
+class AdvancedOptionsDialogState extends State<AdvancedOptionsDialog> {
+  bool _isSelectedOption1 = true;
```
Comment (msesko) [inline packages/io_library/lib/src/ui/advanced_options_dialog.dart]: Name the option that is selected. for example _isAntialiasingSelected

File: packages/io_library/lib/src/ui/advanced_options_dialog.dart [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  bool _isSelectedOption2 = true;
+
+  @override
+  Widget build(BuildContext context) {
+    return AlertDialog(
+      title: const Text(
+        'Advanced Options',
+        style: TextStyle(color: Colors.blue),
```
Comment (msesko) [inline packages/io_library/lib/src/ui/advanced_options_dialog.dart]: Use predefined colors. defined in lightColorScheme

File: packages/io_library/lib/src/ui/advanced_options_dialog.dart [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      content: Column(
+        mainAxisSize: MainAxisSize.min,
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: <Widget>[
+          Row(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: [
+              const Text(
+                'Antialiasing',
+                style: TextStyle(color: Colors.black),
```
Comment (msesko) [inline packages/io_library/lib/src/ui/advanced_options_dialog.dart]: Use predefined colors. defined in lightColorScheme

File: packages/io_library/lib/src/ui/advanced_options_dialog.dart [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+              ),
+              ToggleButtons(
+                isSelected: [_isSelectedOption1],
+                onPressed: (int index) {
+                  setState(() {
+                    _isSelectedOption1 = !_isSelectedOption1;
+                  });
+                },
+                color: Colors.grey,
```
Comment (msesko) [inline packages/io_library/lib/src/ui/advanced_options_dialog.dart]: Use predefined colors. defined in lightColorScheme

File: packages/io_library/lib/src/ui/advanced_options_dialog.dart [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+                fillColor: Colors.blue.shade50,
```
Comment (msesko) [inline packages/io_library/lib/src/ui/advanced_options_dialog.dart]: Use predefined colors. defined in lightColorScheme

File: packages/io_library/lib/src/ui/advanced_options_dialog.dart [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+                ],
+              ),
+            ],
+          ),
+          Row(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: [
+              const Text(
+                'Smoothing',
+                style: TextStyle(color: Colors.black),
```
Comment (msesko) [inline packages/io_library/lib/src/ui/advanced_options_dialog.dart]: Use predefined colors. defined in lightColorScheme

File: packages/io_library/lib/src/ui/advanced_options_dialog.dart [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+              ),
+              ToggleButtons(
+                isSelected: [_isSelectedOption2],
+                onPressed: (int index) {
+                  setState(() {
+                    _isSelectedOption2 = !_isSelectedOption2;
+                  });
+                },
+                color: Colors.grey,
```
Comment (msesko) [inline packages/io_library/lib/src/ui/advanced_options_dialog.dart]: Use predefined colors. defined in lightColorScheme

File: packages/io_library/lib/src/ui/advanced_options_dialog.dart [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+                fillColor: Colors.blue.shade50,
```
Comment (msesko) [inline packages/io_library/lib/src/ui/advanced_options_dialog.dart]: Use predefined colors. defined in lightColorScheme

File: packages/l10n/lib/src/l10n/app_localizations_en.dart [hunk 9]
```diff
@@ -30,4 +30,8 @@ class AppLocalizationsEn extends AppLocalizations {
 
   @override
   String get layers => 'Layers';
+
+  // added advnaced options
```
Comment (msesko) [inline packages/l10n/lib/src/l10n/app_localizations_en.dart]: Is this comment neccesary? Also typos

File: packages/l10n/lib/src/l10n/app_localizations.dart [hunk 10]
```diff
@@ -145,6 +145,9 @@ abstract class AppLocalizations {
   /// In en, this message translates to:
   /// **'Layers'**
   String get layers;
+
+  // Getter for "Advanced Options"
```
Comment (msesko) [inline packages/l10n/lib/src/l10n/app_localizations.dart]: Is comment necessary?

**Review by msesko — CHANGES_REQUESTED** [review]
Hi @vishad-tyagi. Good job on the UI. Check comments for changes. 

Is this ticket UI only? @juliajulie95  Or do you also have ti implement the antialiasing and soothing? Would make sense to me.

+ tests are needed for UI
+ tests for functionality in case it has to be implemented

**Comment by msesko** [comment]
Also state is not saved. If I close the UI and reopen the switches states are not saved.

File: packages/l10n/lib/src/l10n/app_localizations_en.dart (hunk 9 above)
Comment (vishad-tyagi) [reply packages/l10n/lib/src/l10n/app_localizations_en.dart]: I just added comments for quick understanding. I apologise for the typo. I will remove all comments made by me

File: packages/io_library/lib/src/ui/advanced_options_dialog.dart (hunk 2 above)
Comment (vishad-tyagi) [reply packages/io_library/lib/src/ui/advanced_options_dialog.dart]: Ok, I will use the colours from lightColorScheme. For clarity can i add colours to lightColorScheme file or have to work with the colours provided?

File: packages/io_library/lib/src/ui/advanced_options_dialog.dart (hunk 1 above)
Comment (vishad-tyagi) [reply packages/io_library/lib/src/ui/advanced_options_dialog.dart]: I will rename the variables.

**Comment by vishad-tyagi** [comment]
@msesko I worked on the Information provided in the ticket. I commented for more information too. After that, I took as much as I could from the Android version of the app on the Play Store. (I know Catrobat is open source and run by many students, I just don't want you to think, that I did the minimum 🙂). Just let me know what other functionality are to be added. I will commit to all the changes ASAP. Regarding the state save functionality, the specifications for the two options were not clearly defined in the ticket. Consequently, I refrained from incorporating the state change variable, anticipating that this aspect might be addressed in a future ticket.
Sorry for the late response, my mid-sem exams are going on.

**Comment by Lenkomotive** [comment]
> Hi @vishad-tyagi. Good job on the UI. Check comments for changes.
> 
> Is this ticket UI only? @juliajulie95 Or do you also have ti implement the antialiasing and soothing? Would make sense to me.
> 
> * tests are needed for UI
> * tests for functionality in case it has to be implemented

The ticket-description does not clearly state the scope of the ticket. Imo the actual implementation of the algorithms would exceed the scope of the ticket. Id say we treat it as UI only for now and create new tasks for each algorithm to be implemented later.

**Comment by juliajulie95** [comment]
Indeed, I would not want to add UI without working functionality behind it.
We could maybe define the ticket more clearly and implement one of the two options to keep the ticket smaller?

File: packages/io_library/lib/src/ui/advanced_options_dialog.dart (hunk 2 above)
Comment (msesko) [reply packages/io_library/lib/src/ui/advanced_options_dialog.dart]: try to stick with the colors provided in the theme. if necessary you can do adjustments like .withAlpha() and stuff like that.

File: packages/l10n/lib/src/l10n/app_localizations_en.dart (hunk 9 above)
Comment (msesko) [reply packages/l10n/lib/src/l10n/app_localizations_en.dart]: No need to apologize :) I forget comments in my code all the time haha

**Comment by msesko** [comment]
> Indeed, I would not want to add UI without working functionality behind it. We could maybe define the ticket more clearly and implement one of the two options to keep the ticket smaller?

It would be nice to specify if it is UI only ticket or functionality as well for this type of features.

But even if it is a UI feature it should be prepared in a way that the backend functionality can just be plugged into it (no hardcoded vals where we should have vars, correct passing in of vars, correctly managing states etc.)

Such ticket would also be 454 - Layers, where the UI in itself is quite a lot to code and would be easier to review if separated.

**Comment by vishad-tyagi** [comment]
@msesko I have implemented the changes, if any more changes are required let me know (like _showAdvancedOptionsDialog() should be in io_handler instead of overflown_menu, etc)

**Comment by juliajulie95** [comment]
Please resolve conflicts

**Comment by bakicelebi** [comment]
Closing this PR. Create a new PR with the project structure

**Comment by Vainyala** [comment]
Hi..
can i work on this?

**Pull request closed without merging**